python/lsst/resources/s3utils.py

```diff
@@ -230,26 +251,17 @@ def getS3Client(profile: str | None = None) -> boto3.client:
     if not endpoint:
         endpoint = None  # Handle ""
 
-    disable_value = os.environ.get("LSST_DISABLE_BUCKET_VALIDATION", "0")
-    skip_validation = not re.search(r"^(0|f|n|false)?$", disable_value, re.I)
-
-    return _get_s3_client(endpoint, profile, skip_validation)
+    return _parse_endpoint_config(endpoint, profile)
 
 
 @functools.lru_cache
-def _get_s3_client(endpoint: str | None, profile: str | None, skip_validation: bool) -> boto3.client:
+def _get_s3_client(
+    endpoint_config: _EndpointConfig, profile: str | None, skip_validation: bool
+) -> boto3.client:
     # Helper function to cache the client for this endpoint
     config = botocore.config.Config(read_timeout=180, retries={"mode": "adaptive", "max_attempts": 10})
 
-    endpoint_config = _parse_endpoint_config(endpoint)
-
-    if endpoint_config.access_key_id is not None and endpoint_config.secret_access_key is not None:
-        # We already have the necessary configuration for the profile, so do
-        # not pass the profile to boto3.  boto3 will raise an exception if the
-        # profile is not defined in its configuration file, whether or not it
-        # needs to read the configuration from it.
-        profile = None
-    session = boto3.Session(profile_name=profile)
+    session = boto3.Session(profile_name=endpoint_config.profile)
 
```

Comment:
A few lines down (line 273) we run this code to disable bucket name validation in botocore.
```
 if skip_validation:
    client.meta.events.unregister("before-parameter-build.s3", validate_bucket_name)
```

This is apparently needed to make Ceph multi-tenant bucket names work.  Are we still using Ceph multi-tenant buckets?  It looks like fsspec uses s3fs which uses botocore internally, so we may need to figure out how to apply a similar workaround when using fsspec.

Reply:
@ktlim can you remind me of an example place at USDF where we use the names that don't work so I can try this out?

Reply:
@erykoff confirms that a URI such as `"s3://rubin:rubinobs-lfa-cp/Electrometer:101/fits/2024/07/29/Electrometer:101_fits_2024-07-29T17:26:16.810.fits"` fails because fsspec parses the URL and thinks it has a port number. We might have to create an explicit s3fs object instead of using `url_to_fs`.